### DISCUSSION TITLE
Possible to disable cookies when payload too large on the client?

### DISCUSSION THREAD

**Discussion by teamclouday** [socketio/socket.io#5046] [Q&A]
Hi, I'm using socket.io client in nextjs, which connects to a python-socketio server with fastapi.

Today I ran into an issue that the server keeps disconnecting whenever the websocket connection is initialized from client. After many hours of debugging, it turns out that it's because the websockets package has a [limit of request size](https://github.com/python-websockets/websockets/blob/f0398141d2efd28f64d8e1d6d9adc179a9e5e334/src/websockets/legacy/http.py#L197)
Logging the request payload it shows the content of my frontend cookies. (around 9k which is larger than the 8k limit)

Is there anyway to disable sending the cookies when connecting with websocket? Or at least pick what to send? In my case the cookies contain auth tokens from multiple projects. Those cookies won't be used on the server. I tried following code but it doesn't seem to work:
```js
export const socket = io(`${process.env.NEXT_PUBLIC_BACKEND_URL}`, {
  path: "/socket",
  closeOnBeforeunload: false,
  autoConnect: false,
  withCredentials: false,
});
```

**Comment by beegotsy** [comment]
Hi,

have you found a solution?
I am facing the same problem, I have a domain cookie that my socket server doesn't need, but which is needed for other services and therefore cannot be removed.

I would like to make the socket HTTP request without including this (or any) cookie, but it seems impossible.
I am already using the standard [WebSocket API](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket):
```
const socket = new WebSocket(endpoint);
```

From what I understand, the only way to not include the domain cookies is to proxy the socket server to another domain, right?

**Reply by teamclouday** [reply]
Hi unfortunately I didn't find a solution on socketio. But python websockets has made the cookie limit configurable so that's the workaround I'm currently using. https://github.com/python-websockets/websockets/issues/1471
